spec/hyacinth/adapters/preservation_adapter/fedora3_spec.rb

```diff
@@ -0,0 +1,283 @@
+      end
+    end
+  end
+  describe "#generate_new_location_uri" do
+    subject { adapter.generate_new_location_uri }
+    before do
+      allow(pid_generator).to receive(:next_pid).and_return(*([1, 2, 3].map { |x| "test:#{x}" }))
+      allow(adapter).to receive(:exists?).and_return(true, true, false)
+    end
+    it "returns the first new location URI" do
+      is_expected.to eql("fedora3://test:3")
+    end
+  end
+  describe "#persist_impl" do
+    let(:rubydora_object) { Rubydora::DigitalObject.new(object_pid, connection) }
+    let(:hyacinth_object) { DigitalObject::Item.new }
+    let(:profile_xml) { file_fixture("fedora3/new-object.xml").read }
+    let(:datastreams_xml) { file_fixture("fedora3/new-datastreams.xml").read }
+    before do
+      expect(connection).to receive(:find_or_initialize).with(object_pid).and_return(rubydora_object)
+      expect(connection).to receive(:object_profile).with(object_pid, nil).and_return(profile_xml)
+      expect(connection).to receive(:datastreams).with(pid: object_pid, profiles: "true").and_return(datastreams_xml)
+      dsids.each do |dsid|
+        expect(connection).to receive(:datastream_profile).with(object_pid, dsid, nil, nil).and_return({})
+      end
+      expect(rubydora_object).to receive(:save).and_return(nil)
+      allow(rubydora_object).to receive(:relationship).and_return([]) # all new values!
+    end
+    context "core data" do
+      let(:dsids) { [described_class::HYACINTH_CORE_DATASTREAM_NAME, 'structMetadata'] }
+
+      # test rels-ext elsewhere
+      before { allow(connection).to receive(:add_relationship) }
+
+      it "persists core data to Fedora3" do
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        actual_json = JSON.parse(rubydora_object.datastreams[described_class::HYACINTH_CORE_DATASTREAM_NAME].content)
+        expect(actual_json).to have_key("uid")
+      end
+    end
+    context "field exports" do
+      let(:dsids) { ['descMetadata', 'structMetadata'] }
+      let(:digital_object_title) { "Assigned Label" }
+      before do
+        FactoryBot.create(:export_rule) # creates descMetadata
+        expect(connection).to receive(:datastream_profile).with(object_pid, described_class::HYACINTH_CORE_DATASTREAM_NAME, nil, nil).and_return({})
+        # test rels-ext elsewhere
+        allow(connection).to receive(:add_relationship)
+      end
+      it "persists templated field exports to datastreams" do
+        hyacinth_object.dynamic_field_data['name'] = [{ 'role' => "Farmer" }]
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        actual_xml = rubydora_object.datastreams["descMetadata"].content
+        actual_xml.sub!(/^<\?.+\?>/, '') # remove PI
+        actual_xml.gsub!('mods:', '') # remove ns
+        actual_xml.gsub!(/\s/, '') # remove whitespace
+        expect(actual_xml).to eql("<mods><name>Farmer</name></mods>")
+      end
+      it "makes a core object property assignments" do
```

Comment:
See earlier comment about same-name `it "makes a core object property assignments" do` test.